planner/core/physical_plans.go

```diff
@@ -201,44 +202,37 @@ type PhysicalApply struct {
 	rightChOffset int
 }
 
-// PhysicalHashJoin represents hash join for inner/ outer join.
-type PhysicalHashJoin struct {
+type basePhysicalJoin struct {
```

Comment:
If we want to extract common fields of `LogicalJoin` as well, we need to remove `physicalSchemaProducer`, `InnerChildIdx`, `OuterJoinKeys` and `InnerJoinKeys` from the `baesJoin` struct for example. I prefer to just extract `basePhysicalJoin` now... @winoros